Content.Server/Station/Systems/StationSystem.cs

```diff
@@ -0,0 +1,406 @@
+                continue;
+
+            AddGrid(becomesStation.Id, grid);
+        }
+
+        if (!dict.Any())
+        {
+            // Oh jeez, no stations got loaded.
+            // We'll just take the first grid and setup that, then.
+
+            var grid = ev.Grids[0];
+
+            AddGrid("Station", grid);
+        }
+
+        // Iterate over all PartOfStation
+        foreach (var grid in ev.Grids)
+        {
+            if (!TryComp<PartOfStationComponent>(_mapManager.GetGridEuid(grid), out var partOfStation))
+                continue;
+
+            AddGrid(partOfStation.Id, grid);
+        }
+
+        foreach (var (id, gridIds) in dict)
+        {
+            StationConfig? stationConfig = null;
+            if (ev.GameMap.Stations.ContainsKey(id))
+                stationConfig = ev.GameMap.Stations[id];
+            else
+                _sawmill.Error($"The station {id} in map {ev.GameMap.ID} does not have an associated station config!");
+            InitializeNewStation(stationConfig, gridIds.Select(x => _mapManager.GetGridEuid(x)), ev.StationName);
+        }
+    }
+
+    private void OnRoundEnd(GameRunLevelChangedEvent eventArgs)
+    {
+        if (eventArgs.New != GameRunLevel.PreRoundLobby) return;
+
+        foreach (var entity in _stations)
+        {
+            Del(entity);
+        }
+    }
+
+    #endregion Event handlers
+
+
+    /// <summary>
+    /// Generates a station name from the given config.
+    /// </summary>
+    /// <param name="config"></param>
+    /// <returns></returns>
+    public static string GenerateStationName(StationConfig config)
+    {
+        return config.NameGenerator is not null
+            ? config.NameGenerator.FormatName(config.StationNameTemplate)
+            : config.StationNameTemplate;
+    }
+
+    /// <summary>
+    /// Initializes a new station with the given information.
+    /// </summary>
+    /// <param name="stationConfig">The game map prototype used, if any.</param>
+    /// <param name="gridIds">All grids that should be added to the station.</param>
+    /// <param name="name">Optional override for the station name.</param>
+    /// <returns>The initialized station.</returns>
+    public EntityUid InitializeNewStation(StationConfig? stationConfig, IEnumerable<EntityUid>? gridIds, string? name = null)
+    {
+        //HACK: This needs to go in null-space but that crashes currently.
+        var station = Spawn(null, new MapCoordinates(0, 0, _gameTicker.DefaultMap));
+        var data = AddComp<StationDataComponent>(station);
+        var metaData = MetaData(station);
+        data.StationConfig = stationConfig;
+
+        if (stationConfig is not null && name is null)
+        {
+            metaData.EntityName = GenerateStationName(stationConfig);
+        }
+        else if (name is not null)
+        {
+            metaData.EntityName = name;
+        }
+        else
+        {
+            metaData.EntityName = "How did we get here?";
```

Comment:
if this can ever reasonably happen it should probably be given a name that shows its an error